### ISSUE TITLE
Suddenly can't login to certain project anymore

### ISSUE THREAD

**Issue by peXed** [directus/directus#2669]
# Bug Report

## Steps to Reproduce

1. Try to login
2. Look at the network console

## Expected Behavior
Backend is loaded.

## Actual Behavior
"Server Error" gets shown and the API call to: https://mydirectusinstallation.com/users/me?fields[0]=id&fields[1]=avatar.*&fields[2]=email&fields[3]=first_name&fields[4]=last_name&fields[5]=locale&fields[6]=role.*&fields[7]=last_page&fields[8]=theme

Ouputs an 401 error. The response is:
`{
  "error": {
    "code": 3,
    "message": "Unauthorized request",
    "class": "Directus\\Exception\\UnauthorizedException",
    "file": "\/var\/www\/vhosts\/httpdocs\/directus\/src\/core\/Directus\/Services\/UsersService.php",
    "line": 152
  }
} <br/ >
  <b> Notice </b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/ httpdocs/directus/src/core/Directus/Services/ItemsService.php </b> on line <b>557</b> <br/>
`

## Other Context & Screenshots
I did not change anything. The error appeared over night as it seems.
Other projects on the same install are still working perfectly. Only the one project seems broken.

## Technical Details

- Device: Linux Server
- OS: -
- Web Server: -
- PHP Version: 7.4.3
- Database: -
- Directus Release Version: v8.6.2
- Install Method: cloned `master` branch

**Comment by rijkvanzanten** [comment]
Did you upgrade anything on the server that might've caused this? Could it have been an auto-update from your host? 

By the looks of it the fact that PHP plopped on an additional warning breaks the app:

```
...
<br/ > <b> Notice </b>: Trying to access array offset on value of type null in <b>/var/www/vhosts/ httpdocs/directus/src/core/Directus/Services/ItemsService.php </b> on line <b>557</b> <br/> 
```

From a quick search around, people have reported similar issues on some other PHP based projects when upgrading to php 7.4 apparently 🤔

**Comment by peXed** [comment]
@rijkvanzanten Thanks for the reply. I maybe upgrade the PHP version on the server. But should it not work regardless? Also the other projects are still working fine. Only this one project create the error.

**Comment by rijkvanzanten** [comment]
Oh that's weird, yeah I would've expected every project to error out in that case 🤔 

Based on a quick peek in the codebase, it looks like it's failing on reading the status:

Could you check if this user has a status associated with it in the DB? Is the status mapping for directus_users.status in directus_fields setup correctly? Does the status of the user exist in that status mapping?

Normally, those values aren't user editable, but something out of sync there would explain why this might fail

https://github.com/directus/api/blob/68135be20c7cd5df80926c80c20fca722afc9969/src/core/Directus/Services/ItemsService.php#L539-L558

**Comment by peXed** [comment]
@rijkvanzanten Thanks for the reply. I had some time to look into this issue today. The status is correctly set to "active" for all users.
It is still possible to login via the JavaScript-API and get the data.

**Comment by rijkvanzanten** [comment]
Can you check if the server might strip out cookies? The app uses the JS SDK with cookie mode, which is the only difference I can think of

**Comment by peXed** [comment]
The same setup was working perfectly and is working perfectly with other projects.

Is there any other way I can fix this? Copy over the users form another project?

**Comment by rijkvanzanten** [comment]
That could work, sure. I'm just trying to understand why 

> It is still possible to login via the JavaScript-API and get the data.

would work, but regular logins wouldn't I'm almost suspecting that it isn't the users table that is the problem, but one of the other system endpoints that's being fetched.

Could you double check directus_roles and directus_fields to see if the avatar and role exist?

**Comment by peXed** [comment]
I checked the directus_fields table and there is an avatar field and all direcuts_roles fields. The directus_roles table exists as well and is exactly the same as in the other (working) project.

In both brojects, the user is an admin (role 1) and has no avatar (NULL).

**Comment by petingo0** [comment]
Hi, same started happening to us recently to one of our projects. 
If you compare the demo site and our site, we can see that the cookie with the token is not being set anymore and from that point, it starts falling as the app is not able to find the permissions for that user.

![image](https://user-images.githubusercontent.com/64983/81134318-79440500-8fa8-11ea-89e2-9d321a7d488f.png)

**Comment by peXed** [comment]
OK...this is odd.

I checked my installation aswell. For me, the system sets the cookies, but for the "broken" project, it sets a cookie that expired 1 hour ago. For the working project (on the same server, same domain, same everything, except different project), it sets the cookie correctly with an expiration date in an hour.

Is that possible? Is there someware an option for that? The timezone of the user is on both projects the same.

**Comment by peXed** [comment]
I thought about this issue a little bit more. @rijkvanzanten could this issue be linked to daylight saving time? Because it changed on March 29.

**Comment by petingo0** [comment]
Ok, found the problem. Check the timezone you have setup in the server and in PHP. That's the reason the app could be providing expired tokens.

**Comment by peXed** [comment]
@petingo0 I checked the timezone. It is correctly setup. It works for other projects but one. That is why the issue is so strange. Why is it working for some projects and another one is not working? This project is a bit older then the others. Was there a migration in the past, that changed something around timezones and maybe did not work correctly? The timezones for the user are correctly setup as well.

**Comment by peXed** [comment]
@rijkvanzanten @petingo0 I found a fix but maybe there is a bigger problem!

The setting auto_sign_out was set to 60 in the "broken" projekt. If I now set it to 180, it will work (but I will get logged out after 30 minutes and not 180).

I guess the expire date for the cookie does not take the correct timezone into account?

**Comment by petingo0** [comment]
yes, there is some kind of issue, we also had 60 mins for auto_sign_out, we changed that for longer as a workaround. It has some kind of issue.

**Comment by rijkvanzanten** [comment]
What's the timezone on your server? The cookie expiration date is returned in UTC `expires=Mon, 18-May-2020 13:42:57 UTC;`, but if the server isn't in UTC, that apparently is offset?
